internal/olm/client/status.go

```diff
@@ -73,7 +73,9 @@ func (c Client) GetObjectsStatus(ctx context.Context, objs ...runtime.Object) St
 
 func (s Status) HasExistingResources() bool {
 	for _, r := range s.Resources {
-		if r.Resource != nil {
+		// Either the resource was found and returned without error or returned
+		// an existence error.
+		if r.Resource != nil || (r.Error != nil && !apierrors.IsNotFound(r.Error)) {
```

Comment:
If the error is not nil and something like `IsUnauthorized(r.Error)` is true, we would return `true`, right? 

I'm not sure that's right either. A `bool` return value doesn't quite work since we really have three cases to cover.
1. At least one resource exists
2. No resources exist
3. Unknown, we got some unexpected error (basically `r.Error != nil && !apierrors.IsNotFound(r.Error)`)

Reply:
Hmm. It's a little confusing for both of these functions to exist, but that they aren't exact inverses of the others.

WDYT about `func (s *Status) HasInstalledResources() (bool, error)` where error is only ever returned if `r.Error != nil && !apierrors.IsNotFound(r.Error)`